### PR TITLE
SNOW-716364: Revert client app id and add connector regression tests

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-716363

vendor all the connector tests, for future backend SnowflakeSQLAlchemy as client app id tests: https://github.com/snowflakedb/snowflake-sqlalchemy/pull/370

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -99,28 +99,42 @@ def _create_users_addresses_tables_without_sequence(engine_testaccount, metadata
...
+    ],
+)
+def test_connect_args(verify_app_name):
     """
     Tests connect string
 
     Snowflake connect string supports account name as a replacement of
     host:port
     """
+
```
Comment (sfc-gh-aling) [inline tests/test_core.py:137]:
```suggestion
```

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -97,10 +97,52 @@ jobs:
...
+    - name: Upgrade setuptools, pip and wheel
+      run: python -m pip install -U setuptools pip wheel
+    - name: Install tox
+      run: python -m pip install tox
+    - name: List installed packages
+      run: python -m pip freeze
+    - name: Run tests
+      run: python -m tox -e connector_regression --skip-missing-interpreters false
+      env:
+        PYTHON_VERSION: ${{ matrix.python-version }}
```
Comment (sfc-gh-mkeller) [inline .github/workflows/build_test.yml:138]:
```suggestion
```

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[submodule "tests/connector_regression"]
+	path = tests/connector_regression
+	url = git@github.com:snowflakedb/snowflake-connector-python.git
```
Comment (sfc-gh-mkeller) [inline .gitmodules:3]:
```suggestion
	url = git@github.com:snowflakedb/snowflake-connector-python
```

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

**Pull request merged**